bigquery-storage/api/BigQueryStorage.Samples.Tests/BigQueryStorageFixture.cs

```diff
@@ -0,0 +1,98 @@
+﻿/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Google.Api.Gax;
+using Google.Apis.Bigquery.v2.Data;
+using Google.Cloud.BigQuery.V2;
+using Grpc.Core;
+using Xunit;
+
+[CollectionDefinition(nameof(BigQueryStorageFixture))]
+public class BigQueryStorageFixture : IDisposable, ICollectionFixture<BigQueryStorageFixture>
+{
+    private BigQueryClient _client;
+
+    internal List<string> TempDatasetIds { get; } = new List<string>();
+
+    internal string ProjectId { get; }
+
+    public BigQueryStorageFixture()
+    {
+        ProjectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+        if (string.IsNullOrWhiteSpace(ProjectId))
+        {
+            throw new Exception("Missing GOOGLE_PROJECT_ID. You need to set the Environment variable 'GOOGLE_PROJECT_ID' with your Google Cloud Project's project id.");
+        }
+
+        _client = BigQueryClient.Create(ProjectId);
+    }
+
+    /// <summary>
+    /// Ensures the dataset and table with given name and schema exists
+    /// </summary>
+    /// <param name="datasetId">The BigQuery Dataset Id</param>
+    /// <param name="tableName">The BigQuery table name</param>
+    /// <param name="tableSchema">The schema for table</param>
+    internal void EnsureDataSetAndTableExists(string datasetId, string tableName, TableSchema tableSchema)
+    {        
+        BigQueryDataset dataSet = _client.GetOrCreateDataset(datasetId);
+        if (!TempDatasetIds.Contains(datasetId))
+        {
+            TempDatasetIds.Add(datasetId);
+        }
+
+        dataSet.GetOrCreateTable(tableName, tableSchema);
+    }
+
+    /// <summary>
+    ///  Disposes the instance of BigQueryStorageFixture
+    /// </summary>
+    public void Dispose()
+    {
+        try
+        {
+            foreach (string datasetId in TempDatasetIds)
+            {
+                DeleteDataSet(datasetId);
+            }            
+        }
+        catch (RpcException)
+        {
+            // Do nothing, we are deleting on a best effort basis.
+        }
+    }
+
+    /// <summary>
+    ///  Gets the table data rows for the given table name in the BigQuery Dataset.
+    /// </summary>
+    /// <returns></returns>
+    internal async Task<PagedEnumerable<TableDataList, BigQueryRow>> GetTableDataAsync(string datasetId, string tableName)
```

Comment:
I don't think this method is really necessary. The calling code can just use:

```csharp
var rows = _fixture.Client.ListRows(datasetId, tableName);
```

(You'd need to expose the client, of course.)

Reply:
Done.